app/api/views/suppliers.py

```diff
@@ -173,3 +174,68 @@ def get_list():
                         type: string
     """
     return list_suppliers()
+
+
+@api.route('/suppliers/search', methods=['GET'])
+def get_suppliers():
+    """Suppliers search names by keyword
+    ---
+    tags:
+        - suppliers
+    definitions:
+        SupplierMinimal:
+            type: object
+            properties:
+                name:
+                    type: string
+                code:
+                    type: integer
+                panel:
+                    type: boolean
+                sme:
+                    type: boolean
+        Suppliers:
+            type: object
+            properties:
+                sellers:
+                    type: array
+                    items:
+                        $ref: '#/definitions/SupplierMinimal'
+    parameters:
+        - name: keyword
+          in: query
+          type: string
+          required: false
+          description: the keyword to search on
+    responses:
+        200:
+            description: a list of matching suppliers
+            schema:
+                $ref: '#/definitions/Suppliers'
+        400:
+            description: invalid request data, such as a missing keyword param
+    """
+    keyword = request.args.get('keyword') or ''
+    if keyword:
+        results = suppliers.get_suppliers_by_name_keyword(keyword)
+        supplier_results = []
+        for result in results:
+            supplier = {}
+            supplier['name'] = result.name
+            supplier['code'] = result.code
+            supplier['panel'] = False
+            supplier['sme'] = False
+
+            try:
+                supplier['panel'] = 'Digital Marketplace' in [f.framework.name for f in result.frameworks]
+                supplier['sme'] = result.data['seller_type']['sme']
```

Comment:
there could be a chance here where this is never set even though it has a value.